src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -171,6 +194,11 @@ internal void Validate()
                 throw new InvalidOperationException($"A non-empty {nameof(this.HubName)} configuration is required.");
             }
 
+            if (!string.Equals(Environment.GetEnvironmentVariable("WEBSITE_SLOT_NAME"), "Production", StringComparison.InvariantCultureIgnoreCase) && this.isDefaultHubName)
```

Comment:
Two things:
1. In the case where `WEBSITE_SLOT_NAME` is not defined (like when running locally), what is the expected behavior? It seems to me that we will throw even though it's not actually a case where slots are being used.
2. Use `StringComparison.OrdinalIgnoreCase` instead of `StringComparison.InvariantCultureIgnoreCase`. More info [here](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#choosing-a-stringcomparison-member-for-your-method-call).

Reply:
I'll fix both of those, thanks! I'll add a boolean method check that returns true if the slot is production or if it doesn't exist to cover that case.